### PR TITLE
feat: updated fees paid in robotics (FPLA-1676)

### PR DESCRIPTION
**Pull request by KiranVarma02**
**Before creating a pull request make sure that:**

- [x] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###
https://tools.hmcts.net/jira/browse/FPLA-1676

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ *] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java [hunk 1]
```diff
@@ -56,7 +57,7 @@ public RoboticsData prepareRoboticsData(final CaseData caseData, final Long case
         return RoboticsData.builder()
             .caseNumber(caseData.getFamilyManCaseNumber())
             .applicationType(deriveApplicationType(caseData.getOrders()))
-            .feePaid(2055.00)
+            .feePaid(isNotEmpty(caseData.getAmountToPay()) ? toDouble(caseData.getAmountToPay()) : 2055.00)
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:60]: remember that the amountToPay is in pence, is the robot expecting it this way?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ void shouldRemoveTemporaryFieldsWhenPresent() {
             .build());
 
         assertThat(callbackResponse.getData())
-            .doesNotContainKey("amountToPay")
+            .containsEntry("amountToPay", "233300")
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java:120]: This entire test seems pointless now, it is not checking that anything has been removed.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 1 above)
Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:60]: I believe so. @Gogetter, is this right?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java (hunk 2 above)
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java:120]: changed test name to be more meaningful. Let me know what you think!

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 1 above)
Comment (Gogetter) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:60]: @KiranVarma02 we would want to pound value as sending in pence will be wrong.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java [hunk 3]
```diff
@@ -491,6 +492,46 @@ void shouldNotHaveCaseIdPropertyWhenRoboticsDataDeserializes() throws IOExceptio
         }
     }
 
+    @Nested
+    class GetFees {
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:496]: could we have this as `FeePaid`?

Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:496]: done

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 1 above)
Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:60]: thanks @Gogetter for help. Done.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java (hunk 2 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java:120]: maybe something more like should retain payment information as we are not just keeping amountToPay

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java [hunk 4]
```diff
@@ -56,7 +58,7 @@ public RoboticsData prepareRoboticsData(final CaseData caseData, final Long case
         return RoboticsData.builder()
             .caseNumber(caseData.getFamilyManCaseNumber())
             .applicationType(deriveApplicationType(caseData.getOrders()))
-            .feePaid(2055.00)
+            .feePaid(fromCCDMoneyGBP(caseData.getAmountToPay()).map(BigDecimal::doubleValue).orElse(2055.00))
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:61]: Can we keep it as BigDecimal or does it have to be converted to double? I think jackson will serialise it to the same number but would have to check. @Gogetter any suggestions?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java [hunk 5]
```diff
@@ -491,6 +492,46 @@ void shouldNotHaveCaseIdPropertyWhenRoboticsDataDeserializes() throws IOExceptio
...
+        void shouldGetFeePaidFromCaseData() throws JsonProcessingException {
+            caseDataBuilder.amountToPay("100000");
+
+            RoboticsData roboticsData = roboticsDataService.prepareRoboticsData(caseDataBuilder.build(), CASE_ID);
+            String returnedRoboticsJson = roboticsDataService.convertRoboticsDataToJson(roboticsData);
+
+            Map<String, Object> returnedRoboticsDataMap = objectMapper.reader()
+                .forType(new TypeReference<Map<String, Object>>() {
+                })
+                .readValue(returnedRoboticsJson);
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:514]: Can be simplified to `mapper.readValue(returnedRoboticsJson, new TypeReference<>())` I believe

Comment (lewisbirks) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:514]: same for test below as well

Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:514]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java (hunk 2 above)
Comment (KiranVarma02) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerAboutToSubmitTest.java:120]: done

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 4 above)
Comment (Gogetter) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:61]: yes we can make this `BigDecimal` and use `@JsonFormat(shape = JsonFormat.Shape.NUMBER)` annotation on `feePaid` in `RoboticsData`. But the question is, do we want to make this change now and what would the value be?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java [hunk 6]
```diff
@@ -491,6 +492,42 @@ void shouldNotHaveCaseIdPropertyWhenRoboticsDataDeserializes() throws IOExceptio
...
+            caseDataBuilder = prepareCaseData().toBuilder();
+        }
+
+        @Test
+        void shouldGetFeePaidFromCaseData() throws JsonProcessingException {
+            caseDataBuilder.amountToPay("100000");
+
+            RoboticsData roboticsData = roboticsDataService.prepareRoboticsData(caseDataBuilder.build(), CASE_ID);
+            String returnedRoboticsJson = roboticsDataService.convertRoboticsDataToJson(roboticsData);
+
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:510]: do we need to convert to json here? Given we are just asserting result for `feePaid`, could we do 
```java
caseDataBuilder.amountToPay("100000");

RoboticsData roboticsData = roboticsDataService.prepareRoboticsData(caseDataBuilder.build(), CASE_ID);

assertThat(roboticsData.getFeePaid()).isEqualTo(1000.00);
```

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 4 above)
Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:61]: Well the value would be that BigDecimal is how you are supposed to store money in java however I do understand that with the operations we are doing (i.e. none) we won't run into any of the issues that come out of having it as a double

Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java:61]: Should we do any robotics testing if we make this change? I guess we are going to do it anyway!

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java [hunk 7]
```diff
@@ -491,6 +492,42 @@ void shouldNotHaveCaseIdPropertyWhenRoboticsDataDeserializes() throws IOExceptio
...
+            assertThat(returnedRoboticsDataMap).containsEntry("feePaid",1000.00);
+        }
+
+        @Test
+        void shouldGetDefaultFeePaidWhenThereIsNoFeePaidInClaimData() throws JsonProcessingException {
+            caseDataBuilder.amountToPay("");
+
+            RoboticsData roboticsData = roboticsDataService.prepareRoboticsData(caseDataBuilder.build(), CASE_ID);
+            String returnedRoboticsJson = roboticsDataService.convertRoboticsDataToJson(roboticsData);
+
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:523]: same here

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java (hunk 6 above)
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:510]: Agree. Changed it now.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java (hunk 7 above)
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataServiceTest.java:523]: Agree. Changed it now.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java (hunk 4 above)
Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java]: done

**Pull request merged**